ext/mysqli/mysqli.c

```diff
@@ -1224,7 +1227,10 @@ void php_mysqli_fetch_into_hash(INTERNAL_FUNCTION_PARAMETERS, int override_flags
 			}
 			zend_fcall_info_args_clear(&fci, 1);
 		} else if (ctor_params) {
-			zend_throw_exception_ex(zend_ce_exception, 0, "Class %s does not have a constructor hence you cannot use ctor_params", ZSTR_VAL(ce->name));
+			zend_argument_error(zend_ce_exception, ERROR_ARG_POS(3),
```

Comment:
Although... Wouldn't it be possible to still make the default value `[]`. That could mean that the "default constructor" is called if the specified class doesn't have any.

Reply:
If we do that, I think it would be best to only accept `array` and change this check to `ctor_params && zend_hash_num_elements(Z_ARRVAL_P(ctor_params)) != 0`, to make `[]` usable for both no constructor and no arguments, and remove the special NULL value.

This is really what I would prefer, I'm not slightly concerned that it's not consistent with how this argument is treated in other exts -- however, as we only accept an array here right now, it might be better to stick with that.

Reply:
Deal! This is exactly what I thought about! :)